### PR TITLE
Refactoring find methods in controllers

### PR DESCRIPTION
**Pull request by erikmacej**

### CODE REVIEW

File: rest/src/main/java/cz/fi/muni/pa165/rest/controller/MushroomController.java [hunk 1]
```diff
@@ -50,22 +50,22 @@ public MushroomDTO findById(@PathVariable("id") long id) {
...
 
-    @RequestMapping(path = "/find", method = RequestMethod.POST)
+    @RequestMapping(path = "/find", method = RequestMethod.GET)
     public MushroomDTO findByName(@RequestParam("name") String name) {
         return mushroomFacade.findByName(name);
     }
 
-    @RequestMapping(path = "/findbytype", method = RequestMethod.POST)
+    @RequestMapping(path = "/findbytype", method = RequestMethod.GET)
     public List<MushroomDTO> findByType(@RequestBody MushroomType type) {
```
Comment (atiratree) [inline rest/src/main/java/cz/fi/muni/pa165/rest/controller/MushroomController.java]: maybe change it to RequestParam

File: rest/src/main/java/cz/fi/muni/pa165/rest/controller/MushroomController.java [hunk 2]
```diff
@@ -50,22 +50,22 @@ public MushroomDTO findById(@PathVariable("id") long id) {
...
         return mushroomFacade.findByType(type);
     }
 
-    @RequestMapping(path = "/findbydate", method = RequestMethod.POST)
+    @RequestMapping(path = "/findbydate", method = RequestMethod.GET)
     public List<MushroomDTO> findByDate(@Valid @RequestBody DateDTO date) {
```
Comment (atiratree) [inline rest/src/main/java/cz/fi/muni/pa165/rest/controller/MushroomController.java:64]: It should still be POST if it has a RequestBody, or change it to Param

File: rest/src/main/java/cz/fi/muni/pa165/rest/controller/MushroomController.java [hunk 3]
```diff
@@ -50,22 +50,22 @@ public MushroomDTO findById(@PathVariable("id") long id) {
...
         return mushroomFacade.findByDate(date);
     }
 
-    @RequestMapping(path = "/findbydateinterval", method = RequestMethod.POST)
+    @RequestMapping(path = "/findbydateinterval", method = RequestMethod.GET)
     public List<MushroomDTO> findByDate(@Valid @RequestBody DateIntervalDTO interval) {
```
Comment (atiratree) [inline rest/src/main/java/cz/fi/muni/pa165/rest/controller/MushroomController.java:69]: has a RequestBody

File: rest/src/main/java/cz/fi/muni/pa165/rest/controller/VisitController.java [hunk 4]
```diff
@@ -51,22 +51,22 @@ public VisitDTO findById(@PathVariable("id") long id) {
...
 
-    @RequestMapping(path = "/findbymushroom", method = RequestMethod.POST)
+    @RequestMapping(path = "/findbymushroom", method = RequestMethod.GET)
     public List<VisitDTO> findByHMushroom(@RequestParam("id") Long mushroomId) {
         return visitFacade.findByMushroom(mushroomId);
     }
 
-    @RequestMapping(path = "/findbydate", method = RequestMethod.POST)
+    @RequestMapping(path = "/findbydate", method = RequestMethod.GET)
     public List<VisitDTO> findByDate(@Valid @RequestBody DateDTO date) {
```
Comment (atiratree) [inline rest/src/main/java/cz/fi/muni/pa165/rest/controller/VisitController.java:70]: same

File: web/src/main/java/cz/fi/muni/pa165/Initializer.java [hunk 5]
```diff
@@ -2,8 +2,11 @@
 
 import org.springframework.test.context.web.WebAppConfiguration;
 import org.springframework.web.context.request.RequestContextListener;
+import org.springframework.web.filter.CharacterEncodingFilter;
 import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
 
+import javax.servlet.FilterRegistration;
```
Comment (atiratree) [inline web/src/main/java/cz/fi/muni/pa165/Initializer.java]: imports not needed

File: web/src/main/java/cz/fi/muni/pa165/Initializer.java [hunk 6]
```diff
@@ -30,5 +33,6 @@
     public void onStartup(javax.servlet.ServletContext servletContext) throws javax.servlet.ServletException {
         super.onStartup(servletContext);
         servletContext.addListener(RequestContextListener.class);
+
```
Comment (atiratree) [inline web/src/main/java/cz/fi/muni/pa165/Initializer.java:33]: same

Comment (erikmacej) [reply web/src/main/java/cz/fi/muni/pa165/Initializer.java:33]: do you think this servletContext.addListener(RequestContextListener.class); ?

**Pull request merged**